daemon/daemon.c

```diff
@@ -5,6 +5,27 @@
 
 char pidfile[FILENAME_MAX + 1] = "";
 char claimingdirectory[FILENAME_MAX + 1];
+char exepath[FILENAME_MAX + 1];
+
+void get_netdata_execution_path(void)
+{
+    int ret;
+    size_t exepath_size = 0;
+    struct passwd *passwd = NULL;
+    char *user = NULL;
+
+    passwd = getpwuid(getuid());
+    user = (passwd && passwd->pw_name) ? passwd->pw_name : "";
+
+    exepath_size = sizeof(exepath);
```

Comment:
This should be `sizeof(exepath) -1`. There is a `buffer[*size] = '\0';` inside the library.

Reply:
Yeap.